core/src/lightclient/mod.rs

```diff
@@ -354,4 +367,130 @@ impl StarknetClient for StarknetClientImpl {
+                        ..
+                    })
+                    | StarknetTransactionReceipt::L1Handler(L1HandlerTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Declare(DeclareTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Deploy(DeployTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::DeployAccount(
+                        DeployAccountTransactionReceipt {
+                            transaction_hash,
+                            actual_fee,
+                            status,
+                            block_hash,
+                            block_number,
+                            ..
+                        },
+                    ) => {
+                        res_receipt.transaction_hash =
+                            Some(PrimitiveH256::from_slice(&transaction_hash.to_bytes_be()));
+                        res_receipt.gas_used = Some(felt_to_u256(actual_fee));
+                        res_receipt.status_code = match status {
+                            StarknetTransactionStatus::Pending => Some(U64::from(0)),
+                            StarknetTransactionStatus::AcceptedOnL1 => Some(U64::from(1)),
+                            StarknetTransactionStatus::AcceptedOnL2 => Some(U64::from(1)),
+                            StarknetTransactionStatus::Rejected => Some(U64::from(0)),
+                        };
+                        res_receipt.block_hash =
+                            Some(PrimitiveH256::from_slice(&block_hash.to_bytes_be()));
+                        res_receipt.block_number = Some(felt_to_u256(block_number.into()));
+                    }
+                };
+            }
+            MaybePendingTransactionReceipt::PendingReceipt(_) => {
+                return Ok(None);
+            }
+        };
+
+        let starknet_tx = self.client.get_transaction_by_hash(hash_felt).await?;
+        match starknet_tx.clone() {
+            StarknetTransaction::Invoke(invoke_tx) => {
+                match invoke_tx {
+                    InvokeTransaction::V0(v0) => {
+                        res_receipt.contract_address =
+                            Some(starknet_address_to_ethereum_address(v0.contract_address));
```

Comment:
My understanding is that we want to get the EVM address of a Kakarot contract (be it an ExternallyOwnedAccount or a Contract account). At first we simply took the last 20 bytes to represent it as an EVM address. But actually now that I think about it, all Kakarot contracts (EVM smart contracts or accounts) have a view method called `get_evm_address()`

So we should replace occurrences of `starknet_address_to_ethereum_address`(which is slicing) to another function. Let's create an issue for this for now. I'll think about it further and detail more strategy in an issue

Reply:
I stand corrected, it's not that we should replace occurrences of `starnet_address_to_ethereum_address`, we should refactor this function to execute the logic we want it to. Could you create an issue for this specific refactor?